source/geometry_model/initial_topography_model/prm_polygon.cc

```diff
@@ -0,0 +1,157 @@
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    PrmPolygon<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Initial topography");
+        {
+          prm.enter_subsection("Prm polygon");
+          {
+            /**
+             * we need to fill the point lists and topography values. They
+             * are stored in the Topography subsection in the Topography parameter.
+             */
+            std::string temptopo = prm.get("Topography parameters");
+            std::vector<std::string> temp_topographies = Utilities::split_string_list(temptopo,'&');
+            unsigned int temp_topographies_size = temp_topographies.size();
+
+            topography_values.resize(temp_topographies_size,0);
+            point_lists.resize(temp_topographies_size);
+            for (unsigned int i_topo = 0; i_topo < temp_topographies_size; i_topo++)
+              {
+                const std::vector<std::string> temp_topography = Utilities::split_string_list(temp_topographies[i_topo],'>');
+                Assert(temp_topography.size() == 2,ExcMessage ("The given line '" + temp_topographies[i_topo]
+                                                               + "' is not correct. It consists of "
+                                                               + boost::lexical_cast<std::string>(temp_topography.size())
+                                                               + " parts separated by >, but it should only contain "
+                                                               "2 parts: the height and the coordinate, separated "
+                                                               "by a >.'"));
+
+                topography_values[i_topo] = Utilities::string_to_double(temp_topography[0]);
+
+                const std::vector<std::string> temp_coordinates = Utilities::split_string_list(temp_topography[1],';');
+                const unsigned int temp_coordinate_size = temp_coordinates.size();
+                point_lists[i_topo].resize(temp_coordinate_size);
+                for (unsigned int i_coord = 0; i_coord < temp_coordinate_size; i_coord++)
+                  {
+                    const std::vector<double> temp_point = Utilities::string_to_double(Utilities::split_string_list(temp_coordinates[i_coord],','));
+                    Assert(temp_point.size() == 2,ExcMessage ("The given coordinate '" + temp_coordinates[i_coord] + "' is not correct. "
+                                                              "It consists of " + boost::lexical_cast<std::string>(temp_topography.size())
+                                                              + " parts separated by :, but it should only contain 2 parts: "
+                                                              "the longitude and latitude of the the coordinate, separated by a ','."));
```

Comment:
since it's up to other parts of the program to interpret coordinates, maybe say "the two coordinates of the polygon points"?